src/lib/geojson.js

```diff
@@ -84,3 +87,154 @@ export function immutablyReplaceCoordinate(
     ...coordinates.slice(indexes[0] + 1)
   ];
 }
+
+export function flattenPositions(geometry) {
+  let positions = [];
+  if (geometry.type === 'Point') {
+    // positions are not nested
+    positions = [
+      {
+        position: geometry.coordinates,
+        indexes: []
+      }
+    ];
+  } else if (geometry.type === 'MultiPoint' || geometry.type === 'LineString') {
+    // positions are nested 1 level
+    positions = geometry.coordinates.map((position, index) => ({
+      position,
+      indexes: [index]
+    }));
+  } else if (geometry.type === 'Polygon' || geometry.type === 'MultiLineString') {
+    // positions are nested 2 levels
+    for (let a = 0; a < geometry.coordinates.length; a++) {
+      positions = positions.concat(
+        geometry.coordinates[a].map((position, index) => ({
+          position,
+          indexes: [a, index]
+        }))
+      );
+    }
+  } else if (geometry.type === 'MultiPolygon') {
+    // positions are nested 3 levels
+    for (let a = 0; a < geometry.coordinates.length; a++) {
+      for (let b = 0; b < geometry.coordinates[a].length; b++) {
+        positions = positions.concat(
+          geometry.coordinates[a][b].map((position, index) => ({
+            position,
+            indexes: [a, b, index]
+          }))
+        );
+      }
+    }
+  }
+
+  return positions;
+}
+
+// TODO: move these to a test file
+
+function assertEquals(expected, actual, message) {
```

Comment:
Why are there tests in `src/lib/`? Don't they go into a test dir/separate test file?